dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/JdbcEventStore.java

```diff
@@ -1412,7 +1419,10 @@ private StringBuilder dataElementAndFiltersSql(
           String bindParameter = "parameter_" + filterCount;
           int itemType = de.getValueType().isNumeric() ? Types.NUMERIC : Types.VARCHAR;
 
-          if (!de.hasOptionSet()) {
+          if (filter.getOperator().isUnary()) {
```

Comment:
This feels all very convoluted 😅 Not due to your change but already! I wonder how we could simplify this. Should it not be possible to have one function with maybe a switch for the different filters returning the individual expressions of the where clause. Then one mapping all the filters into the where clause 🤷🏻 maybe something to think about for some other day